builtin/providers/aws/resource_aws_lightsail_key_pair.go

```diff
@@ -0,0 +1,214 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/terraform/helper/encryption"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsLightsailKeyPair() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLightsailKeyPairCreate,
+		Read:   resourceAwsLightsailKeyPairRead,
+		Delete: resourceAwsLightsailKeyPairDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			// optional fields
+			"pgp_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			// additional info returned from the API
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// fields returned from CreateKey
+			"fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"private_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			// encrypted fields if pgp_key is given
+			"encrypted_fingerprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"encrypted_private_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsLightsailKeyPairCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lightsailconn
+
+	kName := d.Get("name").(string)
+	var pubKey string
+	var op *lightsail.Operation
+	if pubKeyInterface, ok := d.GetOk("public_key"); ok {
+		pubKey = pubKeyInterface.(string)
+	}
+
+	if pubKey == "" {
+		// creating new key
+		resp, err := conn.CreateKeyPair(&lightsail.CreateKeyPairInput{
+			KeyPairName: aws.String(kName),
+		})
+		if err != nil {
+			return err
+		}
+		if resp.Operation == nil {
+			return fmt.Errorf("[ERR] No operation found for CreateKeyPair response")
+		}
+		if resp.KeyPair == nil {
+			return fmt.Errorf("[ERR] No KeyPair information found for CreateKeyPair response")
+		}
+		d.SetId(kName)
+
+		// private_key and public_key are only available in the response from
+		// CreateKey pair. Here we set the public_key, and encrypt the private_key
+		// if a pgp_key is given, else we store the private_key in state
+		d.Set("public_key", resp.PublicKeyBase64)
+
+		// encrypt private key if pgp_key is given
+		pgpKey, err := encryption.RetrieveGPGKey(d.Get("pgp_key").(string))
```

Comment:
If `strings.HasPrefix` [here](https://github.com/hashicorp/terraform/blob/f-lightsail-key-pair/helper/encryption/encryption.go#L19) is `false`, then `RetrieveGPGKey` will return whatever it got for `pgpKey` as input; in the scenario you describe, thats `""`. So, instead of doing `GetOk` and only going down this chain `if ok`, I just did this. I can change it if you feel strongly enough about it